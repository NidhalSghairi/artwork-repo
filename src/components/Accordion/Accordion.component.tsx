import React from "react";
import MuiAccordion from "@mui/material/Accordion";
import AccordionSummary from "@mui/material/AccordionSummary";
import AccordionDetails from "@mui/material/AccordionDetails";
import Typography from "@mui/material/Typography";

type Props = {
  title: string;
  description: React.ReactElement;
};

const Accordion: React.FC<Props> = ({ title, description }: Props) => {
  return (
    <MuiAccordion>
      <AccordionSummary aria-controls="panel1a-content" id="panel1a-header">
        <Typography>{title}</Typography>
      </AccordionSummary>
      <AccordionDetails>{description}</AccordionDetails>
    </MuiAccordion>
  );
};

export default Accordion;
