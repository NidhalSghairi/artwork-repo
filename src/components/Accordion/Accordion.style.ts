import MuiAccordion from "@mui/material/Accordion";
import styled from "styled-components";

export const StyledMuiAccordion = styled(MuiAccordion)`
  && {
    box-shadow: unset;
    border-bottom: 1px solid;
    width: 90%;
  }
`;
