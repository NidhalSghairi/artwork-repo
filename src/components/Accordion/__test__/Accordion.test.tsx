import React from "react";
import Accordion from "../Accordion.component";
import renderer from "react-test-renderer";

it("should renders correctly", () => {
  const props = {
    title: "test",
    description: <h1> description </h1>,
  };
  const tree = renderer.create(<Accordion {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
