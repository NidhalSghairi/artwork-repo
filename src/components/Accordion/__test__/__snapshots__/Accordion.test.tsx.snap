// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters sc-bcXHqe gLbxmD css-1elwnq4-MuiPaper-root-MuiAccordion-root"
>
  <div
    aria-controls="panel1a-content"
    aria-expanded={false}
    className="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-sh22l5-MuiButtonBase-root-MuiAccordionSummary-root"
    id="panel1a-header"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        test
      </p>
    </div>
  </div>
  <div
    className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
      >
        <div
          aria-labelledby="panel1a-header"
          className="MuiAccordion-region"
          id="panel1a-content"
          role="region"
        >
          <div
            className="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
          >
            <h1>
               description 
            </h1>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should renders correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters sc-bcXHqe gLbxmD css-1elwnq4-MuiPaper-root-MuiAccordion-root"
>
  <div
    aria-controls="panel1a-content"
    aria-expanded={false}
    className="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-sh22l5-MuiButtonBase-root-MuiAccordionSummary-root"
    id="panel1a-header"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        test
      </p>
    </div>
  </div>
  <div
    className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
      >
        <div
          aria-labelledby="panel1a-header"
          className="MuiAccordion-region"
          id="panel1a-content"
          role="region"
        >
          <div
            className="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
          >
            <h1>
               description 
            </h1>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
